src/modules/ekf2/EKF/ekf_helper.cpp

```diff
@@ -643,14 +643,16 @@ void Ekf::updateHorizontalDeadReckoningstatus()
 			}
 		}
 
-		_control_status.flags.inertial_dead_reckoning = true;
-
 	} else {
-		_time_last_horizontal_aiding = _time_delayed_us;
+		if (_time_delayed_us > _params.no_aid_timeout_max) {
+			_time_last_horizontal_aiding = _time_delayed_us - _params.no_aid_timeout_max;
```

Comment:
Did you find this was still necessary? I didn't think it was because isTimedOut() is handling when it's zero and now checking the elapsed time without subtraction.

Reply:
Define necessary ;)
If we want that _valid_timeout_max_ defines the maximum time where we're using inertial dead reckoning then yes. Because after the last measurement, the system is already dead reckoning with IMU data, but we only label it as such after _no_aid_timeout_max_, which is correct.

The question is:
Where does one define the starting point of the inertial dead reckoning? After the last sensor-measurement or after the last sensor timed out.